### PR TITLE
acyclic, bcomps, ccomps, edgepaint, gvcolor, gvpack, mingle, nop, sccmap, unflatten: add page

### PR DESCRIPTION
**Pull request by daniel-birket**
Please see issue #2580.

This PR adds the `graphviz` filter commands: `acyclic`, `bcomps`,
`ccomps`, `edgepaint` (aka `clarify`), `gvcolor` (aka `colorize`),
`gvpack`, `mingle`, `nop`, `sccmap`, `tred`, and `unflatten`. These
commands all accept a graph in gv/dot format, transform it in some way
and output another graph in the same format.

Several of these filters require that the input graph be previously processed by `dot` or another layout engine (without using the -T option) to produce a text `layout.gv` file that includes layout information.

The multiple entries in this PR are substantially similar and were originally generated using `autogen` before manual editing to add unique flags and parameters.

Each filter command's entry includes a list of all Graphviz filters as there is otherwise no easy way for the user to learn the available filters in the package. I suggest that I modify the previously accepted sets of Graphviz layout commands and Graphviz convertor commands to include the word "Graphviz" like this set, instead of just "Layouts" and "Convertors".

- [x] The page (if new), does not already exist in the repository.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

File: pages/common/sccmap.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+> Extract strongly connected components of directed graphs.
+> Graphviz filters: `acyclic`, `bcomps`, `comps`, `edgepaint`, `gvcolor`, `gvpack`, `mingle`, `nop`, `sccmap`, `tred`, & `unflatten`.
+> More information: <https://www.graphviz.org/pdf/sccmap.1.pdf>.
+
+- Extract strongly connected components of directed graphs, printing no statistics:
+
+`sccmap -S {{path/to/input1.gv}} {{path/to/input2.gv}} > {{path/to/output.gv}}`
+
+- Print statistics about the graph, producing no output graph:
```
Comment (marchersimon) [inline pages/common/sccmap.md:11]:
```suggestion
- Print statistics about a graph, producing no output graph:
```

File: pages/common/tred.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# tred
+
+> Compute the transitive reduction of directed graphs.
+> Graphviz filters: `acyclic`, `bcomps`, `comps`, `edgepaint`, `gvcolor`, `gvpack`, `mingle`, `nop`, `sccmap`, `tred`, & `unflatten`.
+> More information: <https://www.graphviz.org/pdf/tred.1.pdf>.
+
+- Compute the transitive reduction of directed graphs:
```
Comment (marchersimon) [inline pages/common/tred.md:7]: Would `generate` also work here? `Compute` doesn't really sound right here.

File: pages/common/unflatten.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+> Adjust directed graphs to improve the layout aspect ratio.
+> Graphviz filters: `acyclic`, `bcomps`, `comps`, `edgepaint`, `gvcolor`, `gvpack`, `mingle`, `nop`, `sccmap`, `tred`, & `unflatten`.
+> More information: <https://www.graphviz.org/pdf/unflatten.1.pdf>.
+
+- Adjust directed graphs to improve the layout aspect ratio:
+
+`unflatten {{path/to/input1.gv}} {{path/to/input2.gv}} > {{path/to/output.gv}}`
+
+- Use `unflatten` as a preprocessor for `dot` layout:
```
Comment (marchersimon) [inline pages/common/unflatten.md:11]: This explaines how the command works, but not really what it does.

File: pages/common/acyclic.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+- Make a directed graph acyclic by reversing some edges:
+
+`acyclic {{path/to/input.gv}} > {{path/to/output.gv}}`
+
+- Print if a graph is acyclic, has a cycle, or is undirected, producing no output graph:
+
+`acyclic -v -n {{path/to/input.gv}}`
+
+- Display help:
```
[the remark this answers is not in the document]
Comment (daniel-birket) [reply pages/common/acyclic.md]: Fixed all.

File: pages/common/bcomps.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# bcomps
+
+> Discompose graphs into their biconnected components.
```
[the remark this answers is not in the document]
Comment (daniel-birket) [reply pages/common/bcomps.md]: Fixed all.

File: pages/common/bcomps.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+> Graphviz filters: `acyclic`, `bcomps`, `comps`, `edgepaint`, `gvcolor`, `gvpack`, `mingle`, `nop`, `sccmap`, `tred`, & `unflatten`.
+> More information: <https://graphviz.org/pdf/bcomps.1.pdf>.
+
+- Discompose graphs into their biconnected components:
```
[the remark this answers is not in the document]
Comment (daniel-birket) [reply pages/common/bcomps.md]: Fixed "one or more" all places.
Not in earlier graphviz layouts and convertors.

File: pages/common/bcomps.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+`bcomps {{path/to/input1.gv}} {{path/to/input2.gv}} > {{path/to/output.gv}}`
+
+- Print the number of blocks and cutvertice, producing no output graph:
```
[the remark this answers is not in the document]
Comment (daniel-birket) [reply pages/common/bcomps.md]: done

File: pages/common/bcomps.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
+# bcomps
+
+> Discompose graphs into their biconnected components.
+> Graphviz filters: `acyclic`, `bcomps`, `comps`, `edgepaint`, `gvcolor`, `gvpack`, `mingle`, `nop`, `sccmap`, `tred`, & `unflatten`.
+> More information: <https://graphviz.org/pdf/bcomps.1.pdf>.
+
+- Discompose graphs into their biconnected components:
+
+`bcomps {{path/to/input1.gv}} {{path/to/input2.gv}} > {{path/to/output.gv}}`
```
[the remark this answers is not in the document]
Comment (daniel-birket) [reply pages/common/bcomps.md]: all places

File: pages/common/gvcolor.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+> Flow colors through a ranked digraph (that was already processed by `dot`).
+> Graphviz filters: `acyclic`, `bcomps`, `comps`, `edgepaint`, `gvcolor`, `gvpack`, `mingle`, `nop`, `sccmap`, `tred`, & `unflatten`.
+> More information: <https://graphviz.org/pdf/gvcolor.1.pdf>.
+
+- Flow colors through a ranked digraph (that was already processed by `dot`):
+
+`gvcolor {{path/to/layout1.gv}} {{path/to/layout2.gv}} > {{path/to/output.gv}}`
+
+- Perform layout, colorization, and output to a picture with one command:
```
[the remark this answers is not in the document]
Comment (daniel-birket) [reply pages/common/gvcolor.md]: dot, neato, and other layout commands primary function is laying out the nodes and edges described in the text file, particularly assigning their positions, which are noted in the updated gv file they produce when they are not converting directly to an image. I don't think the default case of "just do layout" is covered in the tldr `dot.md`. If you include the -T png flag, they skip writing the layout.gv output file and produce a PNG image instead.

File: pages/common/sccmap.md (hunk 1 above)
Comment (daniel-birket) [reply pages/common/sccmap.md]: done

File: pages/common/unflatten.md (hunk 3 above)
Comment (daniel-birket) [reply pages/common/unflatten.md]: There isn't enough room to explain that it adds invisible edges (lines) and adjusts minimum lengths to make a long, thin layout more square. I added "to improve aspect ratio".

**Comment by daniel-birket** [comment]
Some of the above suggested edits can be applied to my earlier graphviz work.

File: pages/common/acyclic.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (marchersimon) [reply pages/common/acyclic.md]: There's no need to. The content of the pages are more important.

File: pages/common/bcomps.md [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+> Decompose graphs into their biconnected components.
+> Graphviz filters: `acyclic`, `bcomps`, `comps`, `edgepaint`, `gvcolor`, `gvpack`, `mingle`, `nop`, `sccmap`, `tred`, & `unflatten`.
+> More information: <https://graphviz.org/pdf/bcomps.1.pdf>.
+
+- Decompose one or more graphs into their biconnected components:
+
+`bcomps {{path/to/input1.gv}} {{path/to/input2.gv ...}} > {{path/to/output.gv}}`
+
+- Print the number of blocks and cutvertice in one or more graphs:
```
Comment (marchersimon) [inline pages/common/bcomps.md:11]:
```suggestion
- Print the number of blocks and cutvertices in one or more graphs:
```

File: pages/common/edgepaint.md [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+
+- Colorize edges of one or more graph layouts (that already have layout information) to clarify crossing edges:
+
+`edgepaint {{path/to/layout1.gv}} {{path/to/layout2.gv ...}} > {{path/to/output.gv}}`
+
+- Colorize edges using a color scheme. (See <https://graphviz.org/doc/info/colors.html#brewer>):
+
+`edgepaint -color-scheme={{accent7}} {{path/to/layout.gv}} > {{path/to/output.gv}}`
+
+- Perform layout, colorization, and output to a picture with one command:
```
Comment (marchersimon) [inline pages/common/edgepaint.md:15]: @sbrl what do you think about this formulation? 
Still not perfect, but maybe an improvement:
```suggestion
- Perform layouting, colorization, and rendering to an image with one command:
```
I don't think `layouting` is a word though.

File: pages/common/tred.md [hunk 12]
```diff
@@ -0,0 +1,13 @@
+# tred
+
+> Compute the transitive reduction of directed graphs.
+> Graphviz filters: `acyclic`, `bcomps`, `comps`, `edgepaint`, `gvcolor`, `gvpack`, `mingle`, `nop`, `sccmap`, `tred`, & `unflatten`.
+> More information: <https://www.graphviz.org/pdf/tred.1.pdf>.
+
+- Generate the transitive reduction of one or more directed graphs:
```
Comment (marchersimon) [inline pages/common/tred.md:7]: @sbrl what do you think about `generate` here? I suggested that and before, it was `compute`. I think none of them really fit, but I can't think of something better.

**Comment by daniel-birket** [comment]
Was there anything else that you wanted here?

**Comment by marchersimon** [comment]
> Was there anything else that you wanted here?

Nope, sorry for the delay.

**Review by marchersimon — APPROVED** [review]
Thanks @daniel-birket

**Pull request merged**